qulice-pmd/src/test/resources/com/qulice/pmd/JunitStaticPublicMethods.java

```diff
@@ -0,0 +1,45 @@
+package foo.test;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public class SomeTest {
+
+    public static final InnerClass INNER = new InnerClass(10, 10);
+
+    @BeforeClass
+    public static void beforeClass(){
+        // setup before class
+    }
+
+    @Test
+    public void emptyTest(){
+        //test something
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        //tear down after class
+    }
+
+    @Parameterized.Parameters
+    public static Collection parameters() {
+        return Collections.EMPTY_LIST;
+    }
+
+    public static class InnerClass {
```

Comment:
@HDouss `StaticPublicMethod` is for negative scenario. Even if there would be mistake in rule implementation and public static fields will raise violation - test set would say nothing. Every tests will be green. 

- What we can do - move public static field and class to `StaticPublicVoidMainMethod`
- Create one more testcase, but I think - it is redundant